pkg/test/harness.go

```diff
@@ -397,9 +397,8 @@ func (h *Harness) RunTests() {
 						t.Fatal(err)
 					}
 
-					tc := report.NewCase(test.Name)
-					test.Run(t, tc)
-					suite.AddTestcase(tc)
```

Comment:
I'm trying to understand why do we need a mutex. It's not clear to me why there's a lock in `NewSubSuite` but not in `AddTestcase` in the previous implementation.

Reply:
AFAICT the old code `AddTestcase` should have used a mutex, since `Testsuite` was shared between funcs that `t.Run(test.Name, ...)` runs, which could run in parallel. So `append`s to `.Testcase` [here](https://github.com/kudobuilder/kuttl/pull/509/files#diff-ad4cc9d2a9b0b703fd754d58e6731bf063ff8e6480905acd5952f90e0bfe7501L151) could have raced.

But maybe there is something that prevented a race that I'm not seeing @kensipe ?

Reply:
I'm convinced that it is the right thing to do, but still curious how it worked in the first place.

Reply:
I guess as long as tests tended not to finish at the exact same time, there was no opportunity for visible breakage.